cronjobs: new filename scheme and G84 added

neteler opened the pull request:
Following the creation of the new GRASS GIS 8.3 release branch an overhaul of the cronjob scripts is needed.

| **label**   | **meaning**                                                     |
| ----------- | --------------------------------------------------------------- |
| legacy      | legacy stable version, no longer recommended for use            |
| old current | current stable version, widely used                             |
| new current | upcoming stable version, for early adopters                     |
| preview     | development version, for developers and new feature enthusiasts |

The labels follow the download sections on https://grass.osgeo.org/download/ (see operating system subpages).

Following this filename scheme, the cronjob files have been renamed and new files been added:

* renamed: `cron_grass7_relbranch_build_binaries.sh -> cron_grass_legacy_build_binaries.sh`
* renamed: `cron_grass7_relbranch_src_snapshot.sh -> cron_grass_legacy_src_snapshot.sh`
* renamed: `cron_grass8_main_build_binaries.sh -> cron_grass_new_current_build_binaries.sh`
* renamed: `cron_grass8_main_src_snapshot.sh -> cron_grass_new_current_src_snapshot.sh`
* renamed: `cron_grass8_relbranch_build_binaries.sh -> cron_grass_old_current_build_binaries.sh`
* renamed: `cron_grass8_relbranch_src_snapshot.sh -> cron_grass_old_current_src_snapshot.sh`
* new file: `cron_grass_preview_build_binaries.sh`
* new file: `cron_grass_preview_src_snapshot.sh`

Importantly, no version numbers are part of the file names to gain long term stability.

In case of version increments, "just" the versions coded in the file(s) need to be updated and (in some cases) new directories been created on grass.osgeo.org - see respective comments in the `cron_grass*.sh` files.

Note: the redundancy in the files may be addressed by refactoring in a separate PR.

neteler committed:
cronjobs: new file name scheme and G84 added

Following the creation of the new GRASS GIS 8.3 release branch an overhaul of the cronjob scripts is needed.

| **label**   | **version number** |
| ----------- | ------------------ |
| legacy      | 7.8                |
| old current | 8.2                |
| new current | 8.3                |
| preview     | 8.4                |

Following this file name scheme, the cronjob files have been renamed and new files been added:

* renamed: cron_grass7_relbranch_build_binaries.sh -> cron_grass7_legacy_build_binaries.sh
* renamed: cron_grass7_relbranch_src_snapshot.sh -> cron_grass7_legacy_src_snapshot.sh
* renamed: cron_grass8_main_build_binaries.sh -> cron_grass8_new_current_build_binaries.sh
* renamed: cron_grass8_main_src_snapshot.sh -> cron_grass8_new_current_src_snapshot.sh
* renamed: cron_grass8_relbranch_build_binaries.sh -> cron_grass8_old_current_build_binaries.sh
* renamed: cron_grass8_relbranch_src_snapshot.sh -> cron_grass8_old_current_src_snapshot.sh
* new file: cron_grass8_preview_build_binaries.sh
* new file: cron_grass8_preview_src_snapshot.sh

Importantly, no version numbers are part of the file names to gain long term stability.

In case of version increments, "just" the versions coded in the file(s) need to be updated and (in some cases) new directories been created on grass.osgeo.org (see respective comments in the `cron_grass*.sh` files).

wenzeslaus reviewed (changes requested):
1. The new names still contain version although "no version numbers are part of the file names" is stated as one of the goals.
2. Same issue, different take on it: The names _grass7_ and _grass8_ go against the _Usage_ section of [Version Numbering RFC](https://github.com/OSGeo/grass/blob/main/doc/development/rfc/version_numbering.md): "Version should be considered a separate item from the name." Or in other words, 7 and 8 are not part the name, but really version numbers.

ninsbl commented:
See also #814 for a more comprehensive change with regards to removing version numbers. May be you can take over / finish of that PR?

wenzeslaus commented on utils/cronjobs_osgeo_lxd/README.md:
```diff
@@ -15,29 +28,31 @@ programmer's manual.
-  * grass8-devel: `cron_grass8_main_src_snapshot.sh`
+  * grass7-stable (legacy): `cron_grass7_legacy_src_snapshot.sh`
+  * grass8-old-current (current stable): `cron_grass8_old_current_src_snapshot.sh`
+  * grass8-new-current (new stable): `cron_grass8_new_current_src_snapshot.sh`
```
If you want to explain somewhere what the strange "old current" means, use the overview table for that instead of trying to fix a cryptic name for each entry in this list.

neteler replied:
Polished in 15f4a5db51b364e20b2af017e695b5434a2a2794: See updated README.md

wenzeslaus replied:
All the pieces except the file name are not necessary, but we can address that later in something like #814.

What I mean is that it is basically just repeating the file name (the thing with dashes) and then the table (the thing in parentheses), so it brings more characters, not more information.

wenzeslaus commented on utils/cronjobs_osgeo_lxd/README.md:
```diff
@@ -15,29 +28,31 @@ programmer's manual.
+  * grass7-stable (legacy): `cron_grass7_legacy_src_snapshot.sh`
+  * grass8-old-current (current stable): `cron_grass8_old_current_src_snapshot.sh`
+  * grass8-new-current (new stable): `cron_grass8_new_current_src_snapshot.sh`
+  * grass8-preview (development): `cron_grass8_preview_src_snapshot.sh`
```
I have no idea where these labels such as `grass8-new-current` appear.

neteler replied:
Polished in 15f4a5db51b364e20b2af017e695b5434a2a2794

wenzeslaus replied:
Where does the thing with dashes appear except for this list?

neteler replied:
It doesn't. It is an "summary" item explanation to not just list the name of the file there.

Better suggestions (please edit directly) are welcome.

wenzeslaus replied:
If I would edit that I would just delete the whole part because the clarity needs to be in the file names not in a possibly outdated list plus-I-don't-get-the-dashes-there, so I pass.

neteler replied:
Ok, removed in 1a3d988fd3c1a45ad7b415ccbe938a0c41062582

wenzeslaus commented on utils/cronjobs_osgeo_lxd/cron_grass8_old_current_src_snapshot.sh:
```diff
@@ -1,13 +1,13 @@
+# script to build GRASS 8 old_current (8.2) source package from the release branch
 # (c) GPL 2+ Markus Neteler <neteler@osgeo.org>
-# Markus Neteler 2002, 2003, 2005, 2006, 2007, 2008, 2012, 2015, 2018-2022
+# Markus Neteler 2002, 2003, 2005, 2006, 2007, 2008, 2012, 2015, 2018-2023
```
I think elsewhere we adopted just a range 2003-2023. Details are in Git or not important.

neteler replied:
Cleaned up in da7d61e51fc0442c3197809530963bef5866569a

wenzeslaus commented on utils/cronjobs_osgeo_lxd/cron_grass8_old_current_src_snapshot.sh:
```diff
@@ -1,13 +1,13 @@
 #
 ## prep
-# git clone https://github.com/OSGeo/grass.git release_branch_8_0
+# git clone https://github.com/OSGeo/grass.git release_branch_8_2
```
What about improving naming of the branches?

neteler replied:
For now too much work as I would have to change everything on the server.

wenzeslaus replied:
What I meant was actually the names of branches in Git which is the likely cause of the inconsistency here. (If it was with underscore in the first place, this would not have happened.)

neteler replied:
Sorry, I don't get it. How are the official branch names related to this PR?

wenzeslaus replied:
The words in them run together which is not desirable, so it leads to inconsistencies like this one when one wants to have the words separate. Nothing to solve in this PR when you need to leave the directory names as is.

wenzeslaus commented on utils/cronjobs_osgeo_lxd/cron_grass8_new_current_src_snapshot.sh:
```diff
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-# script to build GRASS 8.x sources package from the main branch
+# script to build GRASS 8 new_current (8.3) sources package from the main branch
```
The "new current" thing is already on the name, is it necessary to repeat it here with different spelling than in the readme file?

GRASS 8 bla bla (8.3) - see RFC: Version Numbering > Usage > "Version should be considered a separate item from the name. The name of the project and software is GRASS GIS, not GRASS GIS 8." Or in other words, major version is repeated twice here.

neteler replied:
Ok, simplified in da7d61e51fc0442c3197809530963bef5866569a

wenzeslaus replied:
Yes, that's much simpler now!

wenzeslaus commented on utils/cronjobs_osgeo_lxd/README.md:
```diff
@@ -1,5 +1,18 @@
 
+## Version overview
+
+| **label**   | **version number** |
+| ----------- | ------------------ |
+| legacy      | 7.8                |
+| old current | 8.2                |
+| new current | 8.3                |
+| preview     | 8.4                |
```
Perhaps use the table to explain the terms and leave the specific numbers just in the scripts. That way, the table won't need any updates, but it will be actual useful for explaining the reasoning behind the numbers and names.

neteler replied:
Polished in 15f4a5db51b364e20b2af017e695b5434a2a2794

neteler commented:
> See also #814 for a more comprehensive change with regards to removing version numbers. May be you can take over / finish of that PR?

Oh, I wasn't any more aware of that. Not sure if I find the time in the next days to redo everything.

neteler commented:
So I kick this PR.

wenzeslaus commented:
This one is actually closer to what we need to release 8.3.0-RC1, but it depends when we want to release it.

neteler commented:
This PR might be the interim one.

neteler committed:
make cronjob files versionless; update cron_job_list_grass; update README.md

neteler commented:
> 1. The new names still contain version although "no version numbers are part of the file names" is stated as one of the goals.

Fixed.

>     2. Same issue, different take on it: The names _grass7_ and _grass8_ go against the _Usage_ section of [Version Numbering RFC](https://github.com/OSGeo/grass/blob/main/doc/development/rfc/version_numbering.md): "Version should be considered a separate item from the name." Or in other words, 7 and 8 are not part the name, but really version numbers.

Yes, changed.

neteler committed:
some header cleanup

wenzeslaus commented:
It's not clear to me if the line `GPATCH=1 # required by grass-addons-index.sh` needs an update or not.

wenzeslaus commented:
> It's not clear to me if the line `GPATCH=1 # required by grass-addons-index.sh` needs an update or not.

Oh, it's clear to me now. It's for 8.2.1. The others have 0dev as micro as they should.

neteler committed:
remove file explanations

neteler commented:
@wenzeslaus Anything serious yet missing? I'd like to move on with this topic (and the PR is blocked).
